### DISCUSSION TITLE
Datatable pagination, set current page from query string.

### DISCUSSION THREAD

**Discussion by anchan42** [primefaces/community#389] [PrimeVue]
How could I set the current page from a query string?

For example, if a user goes to http://127.0.0.1:8000/index?page=5 using the paginator, page 5 would be highlighted but if they cut and paste the URL in it would not work. The datatable would show the correct data but the current page on the paginator would be 1 which is incorrect.

**Comment by JoseAntonioNavas** [comment]
https://stackoverflow.com/questions/73782037/how-to-programmatically-change-current-page-on-primevue-paginator-component

**Comment by anchan42** [comment]
Thank you. That helps a lot.

This is what I finally did.

- Add v-model:first="offset"  to datatable tag.
- Make offset reactive by const offset = ref(0);
- Then put this code on the onMounted hook.

```
onMounted(() => {
    loadDataFromProp();
    const url = new URL(location.href);
    const page = url.searchParams.get('page');
    offset.value = (dt.value.rows * page ?? 1) - 1;
});
```
